card/mod/01_core/set/all/tracked_attributes.rb

```diff
@@ -79,58 +79,59 @@ def extract_type_id! args={}
     set.reset_set_patterns
 
     if right_id == Card::ReadID && (name_changed? || trash_changed?)
-      self.class.clear_read_rule_cache
-      Card.cache.reset # maybe be more surgical, just Auth.user related
-      expire # probably shouldn't be necessary,
-      # but was sometimes getting cached version when card should be in the
-      # trash.  could be related to other bugs?
-      in_set = {}
-      if !trash && set && (set_class = set.tag)
-        if (class_id =  set_class.id)
-          rule_class_ids = set_patterns.map &:pattern_id
-          # warn "rule_class_id #{class_id}, #{rule_class_ids.inspect}"
-
-          # first update all cards in set that aren't governed by narrower rule
-          Auth.as_bot do
-            cur_index = rule_class_ids.index Card[read_rule_class].id
-            if (rule_class_index = rule_class_ids.index(class_id))
-              set.item_cards(limit: 0).each do |item_card|
-                in_set[item_card.key] = true
-                next if cur_index < rule_class_index
-                if cur_index >= rule_class_index
-                  item_card.update_read_rule
-                end
-              end
-            else warn "No current rule index #{class_id}, " \
-                      "#{rule_class_ids.inspect}"
-            end
-          end
+      update_read_ruled_cards set
+    end
+  end
+end
 
+def update_read_ruled_cards set
+  self.class.clear_read_rule_cache
+  Card.cache.reset # maybe be more surgical, just Auth.user related
+  expire # probably shouldn't be necessary,
+  # but was sometimes getting cached version when card should be in the
+  # trash.  could be related to other bugs?
+  in_set = {}
+  if !trash && set && (set_class = set.tag) && (class_id = set_class.id)
+    rule_class_ids = set_patterns.map &:pattern_id
```

Comment:
Ambiguous block operator. Parenthesize the method arguments if it's surely a block operator, or add a whitespace to the right of the `&` if it should be a binary AND.